Scripts/BuildPhases/ApplyConfiguration.sh

```diff
@@ -1,16 +1,21 @@
 #!/bin/sh
 
+set -e
+
 # The Secrets File Sources
-PRODUCTION_SECRETS_FILE="${PROJECT_ROOT}/.configure-files/WordPress-Secrets.swift"
-INTERNAL_SECRETS_FILE="${PROJECT_ROOT}/.configure-files/WordPress-Secrets-Internal.swift"
-ALPHA_SECRETS_FILE="${PROJECT_ROOT}/.configure-files/WordPress-Secrets-Alpha.swift"
-JETPACK_SECRETS_FILE="${PROJECT_ROOT}/.configure-files/Jetpack-Secrets.swift"
+SECRETS_ROOT="${HOME}/.configure/wordpress-ios/secrets"
+
+PRODUCTION_SECRETS_FILE="${SECRETS_ROOT}/WordPress-Secrets.swift"
+INTERNAL_SECRETS_FILE="${SECRETS_ROOT}/WordPress-Secrets-Internal.swift"
+ALPHA_SECRETS_FILE="${SECRETS_ROOT}/WordPress-Secrets-Alpha.swift"
+JETPACK_SECRETS_FILE="${SECRETS_ROOT}/Jetpack-Secrets.swift"
+
 LOCAL_SECRETS_FILE="${SRCROOT}/Credentials/Secrets.swift"
 EXAMPLE_SECRETS_FILE="${SRCROOT}/Credentials/Secrets-example.swift"
 
 # The Secrets file destination
 SECRETS_DESTINATION_FILE="${BUILD_DIR}/Secrets/Secrets.swift"
```

Comment:
I wonder if it would not be better and more idiomatic to put that file in `$(DERIVED_FILE_DIR)/Secrets.swift` instead (and adjust the project references accordingly), since `DERIVED_FILE_DIR` is the location in which Xcode puts files generated during the build (e.g. CoreData generated classes, Intents generated class files, …) and where it expects us to put these kind of generated files (which it then automatically picks to be compiled)…

Maybe not a big deal as long as we are super-sure that whatever `SECRETS_DESTINATION` we use is `.gitignore`'d and does not risk to be commited, just figured I'd mention what the usual conventions are there 🙂

Reply:
`DERIVED_FILE_DIR` is indeed the location where Xcode expects / suggests to put output files, as one can verify by adding a new file to the list:

<img alt="image" src="https://user-images.githubusercontent.com/1218433/127935105-7a9450c3-0101-49c7-a818-22d8705d7057.png" width=300/>

I didn't do it originally because I didn't know about it 😅 . But, after spending some time trying to make it work, I think it wasn't done like that because of how the project is setup. Namely because developers can read `Secrets.swift` via the file navigator to edit it, which I think is a capability we want to keep for external contributors?

By putting the file in the `BUILD_PRODUCTS` we can reference it in the project using the "Relative to Build Products" location

<img alt="image" src="https://user-images.githubusercontent.com/1218433/127935392-0a458627-2497-485a-a867-6cdc000da5e9.png" width=300/>

Unfortunately, there is not "Relative to Derived Sources" option:

<img alt="image" src="https://user-images.githubusercontent.com/1218433/127935512-a7928f63-eb2b-4de5-8eee-4704c8630312.png" width=300/>

(I'm calling it "Derived Sources" because that's one of the [aliases for `DERIVED_FILE_DIR`](https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html#//apple_ref/doc/uid/TP40003931-CH3-SW43))

It is possible to reference the file in the project by adding a custom location in Preferences > Locations > Custom Paths then update the location for the file:

![image](https://user-images.githubusercontent.com/1218433/127935856-e2dea0d7-5cac-45ee-a8ec-9113c5c5a383.png)

With that, the app compiles but the file is still not accessible, it seems like the path used at build time and the one Xcode computes are different 🤔 

<img width="1657" alt="Screen Shot 2021-08-03 at 10 26 06 am" src="https://user-images.githubusercontent.com/1218433/127939491-8654e741-4c9d-4840-bd3c-1bf41104ad7f.png">

Even if we could make it work, the problem I see with that option is that it would require every developer to make that change in their Xcode setup, and repeat it on every new machine. That sounds like it would generate a lot of support requests. Granted, I'm sure we could find a way to automate it as part of the project bootstrap process, but I wonder if it's worth it.

I also tried bypassing the problem by removing the file reference, but that obviously didn't work because then Xcode didn't know where to find the values defined in `Secrets.swift`.

---

**TL;DR** I wasn't able to make it work in a portable way using `DERIVED_FILE_DIR`. I'm keen on suggestions and improvements because the flow is a bit awkward –even the fact of having the example file tracked in the project is a bit confusing– but I think it's work that might deserve dedicated focus later on.

Reply:
Agreed with all that, good points. Let's keep it the way it is then